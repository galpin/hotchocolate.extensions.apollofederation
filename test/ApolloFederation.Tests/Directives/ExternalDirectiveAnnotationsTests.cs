using System.Threading.Tasks;
using HotChocolate.Types;
using Microsoft.Extensions.DependencyInjection;
using Xunit;
using static HotChocolate.Extensions.ApolloFederation.Test;

namespace HotChocolate.Extensions.ApolloFederation.Directives;

public class ExternalDirectiveAnnotationsTests
{
    [Fact]
    public async Task When_extends_is_specified_on_object()
    {
        var schema = await BuildSchemaAsync(builder =>
        {
            builder.AddObjectType<Product>();
            builder.AddQueryType();
        });

        var sut = schema.GetType<ObjectType>(nameof(Product));

        Assert.Collection(
            sut.Fields["id"].Directives,
            x => Assert.Equal("external", x.Name));
        await schema.QuerySdlAndMatchSnapshotAsync();
    }

    [GraphQLExtends]
    public class Product
    {
        [GraphQLKey]
        [GraphQLExternal]
        public int Id { get; set; }
    }
}