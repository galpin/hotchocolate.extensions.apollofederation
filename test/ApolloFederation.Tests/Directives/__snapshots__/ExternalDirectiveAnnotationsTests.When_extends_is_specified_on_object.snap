{
  "data": {
    "_service": {
      "sdl": "type Product @extends @key(fields: \u0022id\u0022) {\n  id: Int! @external\n}"
    }
  }
}
